core/core-configuration-layer.el

```diff
@@ -196,16 +196,41 @@ If PROPS is non-nil then return packages as lists with their properties"
              :initform nil
              :type boolean
              :documentation
-             "If non-nil this package is excluded from all layers.")))
-
-(defmethod cfgl-package-enabledp ((pkg cfgl-package) &optional inhibit-messages)
+             "If non-nil this package is excluded from all layers.")
+   (depends :initarg :depends
+            :initform nil
+            :type (satisfies (lambda (x) (or (symbolp x) (listp x))))
```

Comment:
Following last comment we could make the init form a list but still accept symbols as well then In `make-package` we transform symbols to lists.